### PR TITLE
BUG: Remove artificial precision limit in rolling var & std

### LINKED ISSUE #37051: DOC/PERF: Decide how to handle floating point artifacts during rolling calculations
Currently we have a check here that artificially handles a numerical precision issue in `rolling.var` and `rolling.std` where our rolling variance calculation is carrying forward floating point artifacts. Ideally we should be using a more numerically stable algorithm (maybe Kahan summation) so this check isn't so arbitrary. 

https://github.com/pandas-dev/pandas/blob/601eff1768380313ec452ba8677cc7e825706796/pandas/_libs/window/aggregations.pyx#L305

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #37051, xref #40448, xref #39872
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/_libs/window/aggregations.pyx [hunk 1]
```diff
@@ -283,10 +283,6 @@ cdef inline float64_t calc_var(int64_t minp, int ddof, float64_t nobs,
             result = 0
         else:
             result = ssqdm_x / (nobs - <float64_t>ddof)
-            # Fix for numerical imprecision.
-            # Can be result < 0 once Kahan Summation is implemented
```
Comment (jbrockmendel) [inline pandas/_libs/window/aggregations.pyx:287]: is there any risk of getting negative variance?

**Review by jreback — APPROVED** [review]
lgtm. just @jbrockmendel question.

File: pandas/_libs/window/aggregations.pyx (hunk 1 above)
Comment (mroeschke) [reply pandas/_libs/window/aggregations.pyx:287]: I don't believe so, no.

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**